urbanairship-adapter/src/main/java/com/gimbal/urbanairship/GimbalAdapter.java

```diff
@@ -360,21 +439,24 @@ public void onAirshipChannelCreated() {
     }
 
     /**
-     * Will automatically call persist the api key and call `Gimbal.setApiKey`
-     * on app start.
-     *
-     * @param gimbalApKey The Gimbal API key.
-     */
-    public void enableGimbalApiKeyManagement(@NonNull String gimbalApKey) {
-        preferences.edit().putString(API_KEY_PREFERENCE, gimbalApKey).apply();
-        Gimbal.setApiKey((Application) context, gimbalApKey);
+     * Set whether the adapter should create a CustomEvent upon Gimbal Place entry.
+     * */
+    public void setShouldTrackCustomEntryEvent(Boolean shouldTrackCustomEntryEvent) {
+        preferences.edit().putBoolean(TRACK_CUSTOM_ENTRY_PREFERENCE_KEY, shouldTrackCustomEntryEvent).apply();
     }
 
     /**
-     * Disables setting the Gimbal API key on app start.
-     */
-    public void disableGimbalApiKeyManagement() {
```

Comment:
Does the removal of this method and `enableGimbalApiKeyManagement` mean we're removing the ability for customers to decide whether `setApiKey` will be called automatically? Do we have any idea what this use-case is, and consequently whether it's alright if we remove it?

Reply:
I don't think so, as `setApiKey` will be called automatically when they call `start()`. I have no idea what the use case for the enable and disable functions were.

Reply:
I think the initialization code is broken for background geofence and beacon app starts, until we make significant changes to our SDK.  At this point the only way I think it works reliably is to first call `enableGimbalApiKeyManagement` and then `start` from `Application.onCreate`. If customer/developers do that, does it even make sense to have the `AirshipReadyReceiver`? Does the API as a whole make sense?